src/ripple/app/hook/impl/applyHook.cpp

```diff
@@ -5398,7 +8137,43 @@ DEFINE_HOOK_FUNCTION(
 
     return hook_float::normalize_xfl(mantissa, exponent, is_negative);
 
-    HOOK_TEARDOWN();
+}
+
+DEFINE_WASM_FUNCTION(
+    int64_t,
+    float_sto_set,
+    uint32_t read_ptr,
+    uint32_t read_len)
+{
+    WASM_HOOK_SETUP();  // populates memory_ctx, memory, memory_length, applyCtx,
+                   // hookCtx on current stack
+
+    if (read_len < 8)
+        return NOT_AN_OBJECT;
+
+    if (NOT_IN_BOUNDS(read_ptr, read_len, memory_length))
+        return OUT_OF_BOUNDS;
+    
+    return __float_sto_set(hookCtx, applyCtx, j, memory + read_ptr, read_len);
+
+    WASM_HOOK_TEARDOWN();
+}
+
+DEFINE_JS_FUNCTION(
+    JSValue,
+    float_sto_set,
+    JSValue raw_sto)
+{
+    JS_HOOK_SETUP();
+
+    auto sto = FromJSIntArrayOrHexString(ctx, raw_sto, 0x10000);
+
+    if (!sto.has_value())
+        returnJS(INVALID_ARGUMENT);
+
+    returnJS(__float_sto_set(hookCtx, applyCtx, j, sto->data(), sto->size()));
```

Comment:
`float_sto_set` should return successful value as bigint(XFL)

```suggestion
    int64_t const out = __float_sto_set(hookCtx, applyCtx, j, sto->data(), sto->size());
    if (out < 0)
        returnJS(out);

    returnJSXFL(out);
```